src/main/java/dataAnalyzer/AllFormations.java

```diff
@@ -0,0 +1,90 @@
+package dataAnalyzer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Calendar;
+
+public class AllFormations {
+    public List<Formations> decideOnFormations(List<FormationResult> flagFormations,
```

Comment:
Why does this function only return `overlappingFormations`? Shouldn't it return all formations except for the ones that overlapped with others. Btw, do we even want to remove overlapping formations? Like, as far as I understand it, that seems to be the whole purpose of this class, but is that even desirable? The user maps each formation they want to a different audio effect anyways, so any overlap would just mean that both audio effects are applied at the same time. I mean, maybe we do want to remove overlaps, then I just wouldn't know about that.

Also, I feel like the idea behind this function is a bit flawed anyways. Why does it accept exactly 3 formations? Because there are 3 kinds of formations? But the user doesn't have to map all formations to some sound parameter. The user could also include formations from several stocks in the mapping.Thus we could have more or less than 3 fomations in any mapping. If we want to remove overlaps (which again, I feel like we might not), then this function should accept either 2 lists of formations or a variable amount of lists of formations imo

Reply:
Thats actually a good point. I figured it would be helpful for the user because they wont that much input but also makes sense to keep the formations. I can definitely throw the class out.